bc_obps/service/form_builder_service.py

```diff
@@ -0,0 +1,196 @@
+    gas_type_one_of = {"gasType": {"oneOf": []}}
+    # For each gas type, add to the enum object to be added to the schema to complete the list of valid gas_types a user can select
+    for index, t in enumerate(gas_types):
+        gas_type_enum.append(t.gas_type.chemical_formula)
+        gas_type_map[t.gas_type.id] = t.gas_type.chemical_formula
+        methodologies = ConfigurationElement.objects.select_related('methodology').filter(
+            reporting_activity_id=activity,
+            source_type_id=source_type,
+            gas_type_id=t.gas_type.id,
+            valid_from__valid_from__lte=report_date,
+            valid_to__valid_to__gte=report_date,
+        )
+        if not methodologies.exists():
+            raise Exception(
+                f'No configuration found for activity_id {activity} & source_type_id {source_type} & gas_type_id {t.gas_type.id} & report_date {report_date}'
+            )
+        # Create the oneOf branch for each gasType selection
+        gas_type_one_of['gasType']['oneOf'].append(
+            {
+                "properties": {
+                    "gasType": {"enum": [t.gas_type.chemical_formula]},
+                    "methodology": {"title": "Methodology", "type": "string", "enum": []},
+                }
+            }
+        )
+
+        methodology_enum = []
+        methodology_one_of = {"methodology": {"oneOf": []}}
+        # For each allowed methodology for the activity-sourceType-gasType relationship, populate the enum with valid methodologies & create the oneOf branch for each methodology selection
+        for u in methodologies:
+            methodology_enum.append(u.methodology.name)
+            methodology_map[u.methodology.id] = u.methodology.name
+            try:
+                methodology_fields = (
+                    ConfigurationElement.objects.prefetch_related('reporting_fields')
+                    .get(
+                        reporting_activity_id=activity,
+                        source_type_id=source_type,
+                        gas_type_id=t.gas_type.id,
+                        methodology_id=u.methodology.id,
+                        valid_from__valid_from__lte=report_date,
+                        valid_to__valid_to__gte=report_date,
+                    )
+                    .reporting_fields.all()
+                )
+            except:
+                raise Exception(
+                    f'No configuration found for activity_id {activity} & source_type_id {source_type} & gas_type_id {t.gas_type.id} & methodology_id {u.methodology.id} & report_date {report_date}'
+                )
+            methodology_object = {"properties": {"methodology": {"enum": [u.methodology.name]}}}
+            # For each field related to a methodology, add that field to the methodology object & append to the oneOf branch
+            for f in methodology_fields:
+                property_field = str_to_camel_case(f.field_name)
```

Comment:
same thing here with the `jsonKey` column. Pretty much anywhere we're using the camel_case function we should consider using something more reliable